pyth_observer/coingecko.py

```diff
@@ -0,0 +1,33 @@
+import json
+import os
+
+from pycoingecko import CoinGeckoAPI
+
+
+ROOT_DIR = os.path.dirname(os.path.abspath(__file__))
+
+cg = CoinGeckoAPI()
+
+
+def get_coingecko_symbol_to_id_mapping():
+    with open(f'{ROOT_DIR}/coingecko_mapping.json') as json_file:
+        data = json.load(json_file)
+
+    return data
+
+
+mapping = get_coingecko_symbol_to_id_mapping()
+
+
+def get_coingecko_prices(symbols):
```

Comment:
this function exposes the coingecko ids. If you re-mapped the symbols -> prices inside this function, then the caller wouldn't have to call get_coingecko_api_id

Reply:
just out of curiosity, what is the concern for exposing the coingecko ids?

Reply:
It's a software engineering thing. There's no reason for the caller to know that these prices even come from coingecko -- they just want a method like `get_prices` that downloads price data for pyth symbols. That method could call coingecko or some other API. If the interface hides how exactly the prices are obtained, it's easy to switch to a different API later. If the interface exposes coingecko-specific stuff, then it's not so easy.

Reply:
got it, thank you for the detailed explanation 🙂